gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -134,18 +135,9 @@ public Fork(TaskContext taskContext, Object schema, int branches, int index) thr
       buildWriterIfNotPresent();
     }
 
-    this.recordQueue = BoundedBlockingRecordQueue.newBuilder()
-        .hasCapacity(this.taskState.getPropAsInt(
+    this.recordQueue = new LinkedBlockingQueue<Object>(this.taskState.getPropAsInt(
```

Comment:
The `BoundedBlockingRecordQueue` class is backed by a `BlockingDeque`; we use the `BlockingDeque.offer(record, timeout, timeunit)` and `BlockingDeque.poll(timeout, timeunit)` methods. This `BlockingDeque` class also has equivalent `put(record)` and `take()` methods that don't require specifying a timeout.

https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/BlockingDeque.html